Api/src/user/user.resolver.ts

```diff
@@ -9,27 +9,38 @@ export class UserResolver {
   constructor(private readonly userService: UserService) {}
 
   @Mutation(() => User)
-  createUser(@Args('createUserInput') createUserInput: CreateUserInput) {
+  createUser(
+    @Args('createUserInput') createUserInput: CreateUserInput,
+  ): Promise<User> {
     return this.userService.create(createUserInput);
   }
 
   @Query(() => [User], { name: 'user' })
-  findAll() {
+  findAll(): Promise<User[]> {
     return this.userService.findAll();
   }
 
   @Query(() => User, { name: 'user' })
-  findOne(@Args('id', { type: () => Int }) id: number) {
+  findOne(@Args('id', { type: () => Int }) id: number): Promise<User> {
     return this.userService.findOne(id);
   }
 
+  @Query(() => User, { name: 'user' })
+  profile(@Args('id', { type: () => Int }) id: number): Promise<User> {
+    return this.userService.findOne(id);
+  }
+
+  // currently not implemented
   @Mutation(() => User)
-  updateUser(@Args('updateUserInput') updateUserInput: UpdateUserInput) {
-    return this.userService.update(updateUserInput.id, updateUserInput);
+  updateUser(
+    @Args('updateUserInput id') updateUserInput: UpdateUserInput,
```

Comment:
I think this is incorrect.

the @args should be per variable. And with options when needed.


```ts
getAuthor(
  @Args('firstName', { nullable: true }) firstName?: string,
  @Args('lastName', { defaultValue: '' }) lastName?: string,
) {}
```
```ts
async getAuthor(@Args('id', { type: () => Int }) id: number) {
  return this.authorsService.findOneById(id);
}
```

Source: https://docs.nestjs.com/graphql/resolvers#args-decorator-options

Reply:
I also just saw that this is done in other modules as well. I think a refactor is needed, to fix previous occurrences.